ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/components/RelationSelector/state/reducer.js

```diff
@@ -0,0 +1,29 @@
+import remove from 'lodash/remove';
+import { SELECT_OPTION, REMOVE_SELECTION, RESET_SELECTIONS } from './types';
+
+export const initialState = {
+  selections: [],
+};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case SELECT_OPTION:
+      let newSelections = [...state.selections];
+      newSelections.push(action.option);
+      return {
+        selections: newSelections,
+      };
+    case REMOVE_SELECTION:
+      let removeSelections = [...state.selections];
+      remove(removeSelections, o => o.metadata.pid === action.removePid);
```

Comment:
I am not sure this is the place to manipulate data.

Reply:
if you know where I can manipulate the state like this I could change it, but I didn't have a better idea
@ntarocco WDYT?

Reply:
@topless what do you mean exactly? The reducer should be the place where to change the state, no?
See [here](https://redux.js.org/basics/reducers/#handling-more-actions).
Do you maybe have something else in mind?